pkg/api/soc.go

```diff
@@ -47,10 +50,11 @@ func (s *Service) socUploadHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	headers := struct {
-		BatchID        []byte        `map:"Swarm-Postage-Batch-Id"`
-		StampSig       []byte        `map:"Swarm-Postage-Stamp"`
-		Act            bool          `map:"Swarm-Act"`
-		HistoryAddress swarm.Address `map:"Swarm-Act-History-Address"`
+		BatchID        []byte            `map:"Swarm-Postage-Batch-Id"`
+		StampSig       []byte            `map:"Swarm-Postage-Stamp"`
+		Act            bool              `map:"Swarm-Act"`
+		HistoryAddress swarm.Address     `map:"Swarm-Act-History-Address"`
+		RLevel         *redundancy.Level `map:"Swarm-Redundancy-Level"`
```

Comment:
I would suggest to use for redundancy level header (`RLevel`), value type instead of a pointer, and all nil checks can be removed. Code will default to `redundancy.NONE` of the header is missing.